scripts/bootstrap/bootstrap-haskell.ps1

```diff
@@ -660,10 +676,22 @@ if (!($DontAdjustBashRc)) {
   $AdjustBashRcExport = 'export BOOTSTRAP_HASKELL_ADJUST_BASHRC=1 ;'
 }
 
+if (!($Msys2Env)) {
+   $SupportedArchitectures = 'AMD64', 'x86'
+  if ($env:PROCESSOR_ARCHITECTURE -eq 'x86') {
+	  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV=MINGW32 ;')
+  } elseif ($env:PROCESSOR_ARCHITECTURE -eq 'AMD64') {
+	  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV=MINGW64 ;')
+  }
+} else {
+  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV={0} ;' -f $Msys2Env)
+  $null = [Environment]::SetEnvironmentVariable("GHCUP_MSYS2_ENV", $Msys2Env, [System.EnvironmentVariableTarget]::User)
+}
+
 if ((Get-Process -ID $PID).ProcessName.StartsWith("bootstrap-haskell") -Or $InBash) {

```

Comment:
I'm confused here as well about why this is done. Please add a code comment to clarify why this is needed.